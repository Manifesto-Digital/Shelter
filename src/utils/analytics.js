export const sendEvent = data => {
  window.dataLayer.push(data);
};
