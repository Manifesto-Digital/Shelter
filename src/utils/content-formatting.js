export const addMarksToContent = content => {
  const markup = content
    .map(block => {
      const { marks, value } = block
      // As there may be multiple marks per piece of content, we need to
      // store the opening and closing tags and then wrap the content
      if (marks && marks.length > 0) {
        let openingMarks = ''
        let closingMarks = ''

        marks.forEach(mark => {
          if (mark.type === 'underline') {
            openingMarks = `<span className="underline">${openingMarks}`
            closingMarks = `${closingMarks}</span>`
          }
          if (mark.type === 'italic') {
            openingMarks = `<em>${openingMarks}`
            closingMarks = `${closingMarks}</em>`
          }
          if (mark.type === 'bold') {
            openingMarks = `<strong>${openingMarks}`
            closingMarks = `${closingMarks}</strong>`
          }
        })
        return `${openingMarks}${value}${closingMarks}`
      }
      return value
    })
    .join('')

  return markup
}
