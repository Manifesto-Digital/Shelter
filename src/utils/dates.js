import dayjs from 'dayjs';

// Format date as string in required format
// Formatting options: https://bit.ly/2R9uVsh
export const dateAsString = (date, format) =>
  dayjs(date)
    .format(format)
    .toString();

export const getWeekDays = () => {
  // Base date is a Monday, this ensures we get the day lsit in order
  const baseDate = new Date(Date.UTC(2019, 0, 7));
  const weekDays = [];
  for (let i = 0; i < 7; i += 1) {
    weekDays.push(baseDate.toLocaleDateString('en-GB', { weekday: 'long' }));
    baseDate.setDate(baseDate.getDate() + 1);
  }
  return weekDays;
};

export const dayOfTheWeek = () => {
  const daysArray = getWeekDays();
  const today = new Date();

  return daysArray[today.getDay() - 1];
};
