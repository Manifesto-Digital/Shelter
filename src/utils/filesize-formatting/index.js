/**
 * Formats integer to human readable filesize
 *
 * @param {int} bytes
 * @param {int} decimalPoint
 */
export const formatFilesize = (bytes, decimalPoint) => {
  if (bytes === undefined) throw new Error('Bytes are required');
  if (bytes < 0) throw new Error('Negative bytes not allowed');
  if (bytes >= 10 ** 27)
    throw new Error(`Bytes too large, must be < 10 ** 27 (${10 ** 27})`);

  if (bytes === 0) return '0 Bytes';

  const k = 1000;
  const dm = decimalPoint || 2;
  const sizes = ['Bytes', 'KB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB'];
  const i = Math.floor(Math.log(bytes) / Math.log(k));

  const sizeToReturn = `${parseFloat((bytes / k ** i).toFixed(dm))} ${
    sizes[i]
  }`;
  return sizeToReturn;
};
