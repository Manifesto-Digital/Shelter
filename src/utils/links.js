export const getInternalLink = slug => {
  const baseURL = window.location.origin
  return `${baseURL}/${slug}`
}
