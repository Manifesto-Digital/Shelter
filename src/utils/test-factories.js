// We define some factories for creating the various types of objects in our system.
// These are just functions that take `Partial<T>` and return `T`, assigning defaults to
// the properties that aren't specified.

export const createFactory = defaults => properties => ({
  ...defaults,
  ...properties,
});

export const createInternalLink = createFactory({
  id: 'ee6c2ca6-54d8-5c26-bbff-7dafdaa823e3',
  title: 'Test Page Title',
  slug: 'internal-link-slug',
});

export const createImage = createFactory({
  id: 'ca8e4a9c-9491-5257-8e53-d3d74ae204a5',
  description: 'Collection hero',
  title: 'collection-hero',
  file: {
    url:
      '//images.ctfassets.net/6sxvmndnpn0s/2DPKnmx9Na8WYgG4ySqkA/5ca89770eb1ac36c9dbfe34d8d65eb5c/collection-hero.jpg',
    fileName: 'collection-hero.jpg',
    contentType: 'image/jpeg',
  },
});

export const createVideo = createFactory({
  id: 'ca8e4a9c-9491-5257-8e53-d3d74ae204a5',
  title: 'Urban Rush hero',
  description: 'Urban rush video',
  file: {
    url:
      '//videos.ctfassets.net/6sxvmndnpn0s/5hlfsk9dwVVNBlbpWHSLvK/9f1d647e6332be389465c9d014a57a26/shelter-urban-rush.mp4',
    fileName: 'shelter-urban-rush.mp4',
    contentType: 'video/mp4',
  },
});

export const createChildContentfulRichText = createFactory({
  json: {
    data: {},
    content: [
      {
        data: {},
        content: [
          {
            data: {},
            marks: [],
            value: 'Some body text',
            nodeType: 'text',
          },
        ],
        nodeType: 'paragraph',
      },
      {
        data: {},
        content: [
          {
            data: {},
            marks: [],
            value: 'Some header text',
            nodeType: 'text',
          },
        ],
        nodeType: 'heading-2',
      },
      {
        data: {},
        content: [
          {
            data: {},
            marks: [
              {
                type: 'bold',
              },
            ],
            value: 'some bollld text',
            nodeType: 'text',
          },
        ],
        nodeType: 'paragraph',
      },
    ],
  },
  childContentfulRichText: {
    html:
      '<p>Our advice and support services across the UK give people <b>one-to-one</p>',
  },
});

export const createFile = createFactory({
  description: 'Mock file description',
  title: 'Mock file title',
  file: {
    contentType: 'application/pdf',
    url:
      '//assets.ctfassets.net/6sxvmndnpn0s/6gDmPHZ04gCMMS4sOa0IEk/7c712c5e8e383bbafd61f2d9d570de4f/component_tree-pdf.pdf',
    fileName: 'component tree-pdf.pdf',
    details: {
      size: 78019,
      image: null,
    },
  },
});

export const createHeaderNavigation = createFactory({
  id: 'e230d8b8-4ee6-5d4c-bf25-57af664d12d7',
  name: 'Assembly - Primary Navigation',
  additionalLink: [
    {
      id: 'e41df6b6-52e2-52db-93c7-9d7ecef7fcd4',
      title: 'Tobys Page',
      slug: 'tobys-page',
    },
  ],
  menuLabel: 'What we do',
  navigationItems: [
    {
      id: 'd582da1f-1f7f-5f4c-ae3b-ef2b41972dbc',
      menuLabel: 'Housing Advice',
      navigationLink: [{ title: 'Tobys Page', slug: 'tobys-page' }],
      subNavigationItems: [
        {
          slug: '320,000-people-in-britain-are-now-homeless',
          title: '320,000 people in Britain are now homeless',
        },
        {
          title: 'Tobys Page',
          slug: 'tobys-page',
        },
        {
          title: 'Shelter Manchester to hold vigil for lost homeless people',
          slug: 'shelter-manchester-to-hold-vigil-for-lost-homeless-people',
        },
        {
          title: 'Shelter Demo Page',
          slug: 'shelter-demo-page',
        },
      ],
    },
  ],
  footerText: createChildContentfulRichText(),
});

export const createSubpage = createFactory({
  id: '27757b6a-db0e-5aca-8db2-8f3e736b5341',
  slug: 'advice-guide-test',
  title: 'Advice guide test',
  subPages: null,
});

export const createContentCards = createFactory({
  pageInformation: {
    pageThumbnail: {
      description: 'Telli Afrik and family',
      file: {
        url:
          '//images.ctfassets.net/6sxvmndnpn0s/3lzXpdUbIjH5rj…bf633ef/Telli_Afrik_and_family_-_Steve_Franck.jpg',
        fileName: 'Telli Afrik and family - Steve Franck.jpg',
      },
    },
    shortDescription: {
      shortDescription: 'A short description for Callams test page',
    },
  },
  slug: 'shelter-demo-page',
  title: 'Shelter Demo Page',
});

export const createContentCardBanner = createFactory({
  header: createChildContentfulRichText(),
  bannerColour: 'Grey',
  bannerFlow: 'Horizontal',
  contentCards: [createContentCards(), createContentCards()],
});

export const createTestRef = createFactory({
  current: {},
});

export const createEvent = createFactory({
  eventName: 'Brighton Half Marathon',
  shortDescription:
    'Run for charity with Team Shelter; half marathons, marathons and other distance running events all in the name of a good cause.',
  thumbnailImage: createImage(),
  eventType: 'Run',
  eventStatus: 'Sold Out',
  eventSystemDate: '2019-02-24',
  eventDisplayDate: '24 February 2019',
  distance: '13.1 miles',
  registrationFee: 19,
  pledge: 199,
  eventLocation: {
    lat: 50.8374204,
    lon: -0.1061897,
  },
  displayLocation: 'Brighton',
  notifiedTeamEmail: 'running@shelter.org.uk',
});

export const createPerson = createFactory({
  id: 'd5ec5ab5-6824-5488-b9bc-4d8794fba4c6',
  firstName: 'Toby',
  lastName: 'Bushell',
  phoneNumber: '012345678910',
  emailAddress: 'test@test.com',
  photo: createImage(),
  bio: {
    internal: {
      content:
        'Nunc nonummy metus. Fusce pharetra convallis urna. Praesent metus tellus, elementum eu, semper a, adipiscing nec, purus.',
    },
  },
  jobTitle: 'Developer',
});

export const createInternalRef = createFactory({
  internal: {
    type: 'ContentfulPageAssemblyContentPage',
  },
  slug: 'internal-link-slug',
});

export const createExternalRef = createFactory({
  internal: {
    type: 'ContentfulTopicExternalLink',
  },
  URL: 'http://mock.com',
  newTab: false,
});
