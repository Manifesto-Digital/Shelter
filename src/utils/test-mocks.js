// Whenever LinkHandler is used, when mounted in enzyme the rendered HTML
// includes a lowercase "l" in the gatsby Link tag. Which causes a warning
// This suppresses that warning for a whole test file
export const hidePascalCaseWarning = () => {
  beforeEach(() => {
    console.error = jest.fn(error => {
      if (
        !error.includes(
          'using incorrect casing. Use PascalCase for React components, or lowercase for HTML elements'
        )
      ) {
        throw new Error(error);
      }
    });
  });

  afterEach(() => {
    console.error.mockClear();
  });
};
