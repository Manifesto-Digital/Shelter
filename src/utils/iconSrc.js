import SpeechBubble from '../assets/svg/icons/comment-light.svg';
import Envelope from '../assets/svg/icons/envelope-light.svg';
import Download from '../assets/svg/icons/download-light.svg';
import Printer from '../assets/svg/icons/print-light.svg';
import OpenBook from '../assets/svg/icons/book-open-light.svg';
import Exclamation from '../assets/svg/icons/exclamation-circle-light.svg';
import Telephone from '../assets/svg/icons/phone-light.svg';
import MapMarker from '../assets/svg/icons/map-marker-alt-light.svg';
import ChippedTimeResults from '../assets/svg/icons/challenge-event/chipped-time-results.svg';
import FinishersMedal from '../assets/svg/icons/challenge-event/finishers-medal.svg';
import FreeBagTransfer from '../assets/svg/icons/challenge-event/free-bag-transfer.svg';
import FreeRacePhotos from '../assets/svg/icons/challenge-event/free-race-photos.svg';
import GoodieBags from '../assets/svg/icons/challenge-event/goodie-bags.svg';
import PostRaceCelebration from '../assets/svg/icons/challenge-event/post-race-celebration.svg';
import SportsMassage from '../assets/svg/icons/challenge-event/sports-massage.svg';
import TechnicalRunningTop from '../assets/svg/icons/challenge-event/technical-running-top.svg';
import ClipboardChecklist from '../assets/svg/icons/clipboard-checklist-light.svg';
import FundingGoalMeter from '../assets/svg/icons/funding-goal-meter-light.svg';
import Lightbulb from '../assets/svg/icons/lightbulb-light.svg';
import MapUnfolded from '../assets/svg/icons/map-unfolded-light.svg';

import { consistentString } from './content-formatting';

const iconList = {
  'speech-bubble': SpeechBubble,
  envelope: Envelope,
  download: Download,
  printer: Printer,
  'open-book': OpenBook,
  'exclamation-mark': Exclamation,
  telephone: Telephone,
  'map-marker': MapMarker,
  'chipped-time-results': ChippedTimeResults,
  'finishers-medal': FinishersMedal,
  'free-bag-transfer': FreeBagTransfer,
  'free-race-photos': FreeRacePhotos,
  'goodie-bags': GoodieBags,
  'post-race-celebration': PostRaceCelebration,
  'sports-massage': SportsMassage,
  'technical-running-top': TechnicalRunningTop,
  'clipboard-checklist': ClipboardChecklist,
  'funding-goal-meter': FundingGoalMeter,
  lightbulb: Lightbulb,
  'map-unfolded': MapUnfolded,
  'no-icon': null,
  default: null,
};

// returns the src of an svg icon from the above icon list
const iconSrc = icon => {
  const iconKey = consistentString(icon);

  if (iconList[iconKey] === undefined) {
    throw new Error(
      'No matching icon was found. Is it set as an option in the CMS? Has it been imported and included in the object above?'
    );
  }

  const chosenIcon = iconList[iconKey] || null;

  if (!chosenIcon) return null;

  return chosenIcon;
};

export default iconSrc;
