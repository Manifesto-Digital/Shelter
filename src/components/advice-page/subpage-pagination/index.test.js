import React from 'react';
import { snapshotComponent, mountWithTheme } from 'test-helpers';
import { createFactory, createSubpage } from '../../../utils/test-factories';
import { hidePascalCaseWarning } from '../../../utils/test-mocks';
import SubPagePagination from '.';
import LinkHandler from '../../link-handler';

const createSubPagePagination = createFactory({
  title: 'An example of a advice guide',
  pages: [
    createSubpage({ slug: 'Advice page 1' }),
    createSubpage({ slug: 'Advice page 2' }),
    createSubpage({ slug: 'Advice page 3' }),
    createSubpage({ slug: 'Advice page 4' }),
  ],
});

it('renders correctly', () => {
  const mockData = createSubPagePagination();
  snapshotComponent(
    <SubPagePagination
      subpages={mockData}
      activeSlug={mockData.pages[2].slug}
    />
  );
});

hidePascalCaseWarning();

it('shows previous and next navigation items if possible', () => {
  const mockData = createSubPagePagination();
  const wrapper = mountWithTheme(
    <SubPagePagination
      subpages={mockData}
      activeSlug={mockData.pages[1].slug}
    />
  );

  expect(wrapper.find(LinkHandler)).toHaveLength(2);
});

it('shows only next if previous not possible', () => {
  const mockData = createSubPagePagination();
  const wrapper = mountWithTheme(
    <SubPagePagination
      subpages={mockData}
      activeSlug={mockData.pages[0].slug}
    />
  );

  expect(wrapper.find(LinkHandler)).toHaveLength(1);
  expect(wrapper.find(LinkHandler).props().type).toEqual('next');
});

it('shows only previous if next not possible', () => {
  const mockData = createSubPagePagination();
  const subpagesLength = mockData.pages.length;
  const wrapper = mountWithTheme(
    <SubPagePagination
      subpages={mockData}
      activeSlug={mockData.pages[subpagesLength - 1].slug}
    />
  );
  expect(wrapper.find(LinkHandler)).toHaveLength(1);
  expect(wrapper.find(LinkHandler).props().type).toEqual('prev');
});
