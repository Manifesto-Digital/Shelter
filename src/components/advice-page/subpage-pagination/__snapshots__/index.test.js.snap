// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c3 {
  -webkit-text-decoration: none;
  text-decoration: none;
  margin-right: auto;
}

.c3 span {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  font-weight: normal;
}

.c6 {
  -webkit-text-decoration: none;
  text-decoration: none;
  margin-left: auto;
}

.c6 span {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  font-weight: normal;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  font-size: 1.4rem;
  margin: 0 0 0.4em;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  font-size: 1.4rem;
  margin: 0 0 0.4em;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 40px;
  padding-bottom: 40px;
  border-bottom: 2px solid #f7f7f7;
}

.c5 svg {
  display: block;
  width: 35px;
  height: 40px;
}

.c0 {
  max-width: 1140px;
  margin: 0 auto;
  padding: 0 20px;
}

.c1 {
  max-width: 100%;
  width: 100%;
  padding: 0 0;
}

@media (min-width:40.0625em) {
  .c1 {
    max-width: 66%;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <Link
        className="c3"
        to="/Advice page 2"
      >
        <div
          className="c4"
          type="prev"
        >
          Prev
          <span
            className="isvg loading c5"
          />
        </div>
        <span>
          Advice guide test
        </span>
      </Link>
      <Link
        className="c6"
        to="/Advice page 4"
      >
        <div
          className="c7"
          type="next"
        >
          Next
          <span
            className="isvg loading c5"
          />
        </div>
        <span>
          Advice guide test
        </span>
      </Link>
    </div>
  </div>
</div>
`;
